tests/test_http.py

```diff
@@ -476,6 +496,110 @@ def _is_server_running(cls, port: int) -> bool:
                 return False
 
 
+class HttpResourcePathConfigTestCase(unittest.TestCase):
+    """Test for the HttpResourcePathConfig class."""
+
+    def test_send_expect_header(self):
+        # Ensure environment variable LSST_HTTP_PUT_SEND_EXPECT_HEADER is
+        # inspected to initialize the HttpResourcePath config class.
+        os.environ.pop("LSST_HTTP_PUT_SEND_EXPECT_HEADER", None)
```

Comment:
In theory a test should not change the environment permanently in a way that might affect other tests.  It looks like below you use `unittest.mock.patch.dict()` with `clear=True`. Couldn't you also use that approach here?

```python
with unittest.mock.patch.dict(os.environ, {}, clear=True):
    config = HttpResourcePathConfig()
    self.assertFalse(config.send_expect_on_put)
```
?